features/storage/TESTS/kvstore/direct_access_devicekey_test/main.cpp

```diff
@@ -268,15 +268,15 @@ void test_direct_access_to_device_inject_root()
         internal_start_address =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS;
         internal_rbp_size =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE;
     } else if (strcmp(STR(MBED_CONF_STORAGE_STORAGE_TYPE), "default") == 0) {
-#if COMPONENT_QSPIF || COMPONENT_SPIF || COMPONENT_DATAFLASH
+#if COMPONENT_FLASHIAP
+        internal_start_address =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_BASE_ADDRESS;
+        internal_rbp_size =  MBED_CONF_STORAGE_TDB_INTERNAL_INTERNAL_SIZE;
+#elif COMPONENT_QSPIF || COMPONENT_SPIF || COMPONENT_DATAFLASH
```

Comment:
I would say this configuration isn't even possible without FLASHIAP - please see [TDB_External documentation](https://github.com/ARMmbed/mbed-os/blob/00f14b7975eeb2b3291d571230ad857141c0fad1/docs/design-documents/features/storage/Configuration/CONFIGURATION.md#tdb_external). What I'm saying is that the TDB_External option should be dropped from default configurations. Only possibility would be to use [TDB_External_no_RBP configuration](https://github.com/ARMmbed/mbed-os/blob/00f14b7975eeb2b3291d571230ad857141c0fad1/docs/design-documents/features/storage/Configuration/CONFIGURATION.md#tdb_external_no_rbp)

Reply:
Removed @VeijoPesonen please check.